p2p/sync.go

```diff
@@ -370,6 +359,16 @@ func (s *syncService) adaptAndSanityCheckBlock(ctx context.Context, header *spec
 				StateDiff: p2p2core.AdaptStateDiff(stateReader, contractDiffs, classes),
 			}
 
+			newClasses := make(map[felt.Felt]core.Class, len(classes))
+			for _, class := range classes {
+				hash, err := class.Hash()
+				if err != nil {
+					bodyCh <- blockBody{err: fmt.Errorf("failed to calculate class hash: %v", err)}
+					return
+				}
+				newClasses[*hash] = class
+			}
+
```

Comment:
These changes could be done above. Moving the changes further creates a diff that makes it difficult to see what changed.